### PR TITLE
docs: add ts demo for extending-a-theme

### PR DESCRIPTION
**Pull request by YunYouJun**
I'm sorry for my carelessness.

> But I remember my configuration pasted from the document and it used to seem like 'extend'. I thought it might appear in changelog, but I didn't find it.

Perhaps typescript type tip is a better idea to use them as examples in a document.

### CODE REVIEW

File: docs/zh/advanced/cookbook/extending-a-theme.md [hunk 1]
```diff
@@ -19,12 +22,38 @@ module.exports = {
 }
 ```
 
+  </CodeGroupItem>
+
+  <CodeGroupItem title="TS">
+
+```ts
+import { path } from '@vuepress/utils'
+import { ThemeObject } from '@vuepress/core'
```
Comment (meteorlxy) [inline docs/zh/advanced/cookbook/extending-a-theme.md:31]: `core` should put before `utils`

Comment (meteorlxy) [inline docs/zh/advanced/cookbook/extending-a-theme.md:31]: `import type`

File: docs/zh/advanced/cookbook/extending-a-theme.md [hunk 2]
```diff
@@ -19,12 +22,38 @@ module.exports = {
...
+export default {
+  // 你的主题
+  name: 'vuepress-theme-foo',
+  // 要继承的父主题
+  extends: 'vuepress-theme-bar',
+  // 覆盖父主题的布局
+  layouts: {
+    Layout: path.resolve(__dirname, 'layouts/Layout.vue'),
+  },
+} as ThemeObject
```
Comment (meteorlxy) [inline docs/zh/advanced/cookbook/extending-a-theme.md:42]: Should be consistent with the later snippet, i.e. `const fooTheme ...`

**Review by meteorlxy — CHANGES_REQUESTED** [review]

File: docs/zh/advanced/cookbook/extending-a-theme.md (hunk 2 above)
Comment (YunYouJun) [reply docs/zh/advanced/cookbook/extending-a-theme.md]: Fixed.

File: docs/advanced/cookbook/extending-a-theme.md [hunk 3]
```diff
@@ -19,12 +22,40 @@ module.exports = {
...
 
+  </CodeGroupItem>
+
+  <CodeGroupItem title="TS">
+
+```ts
+import type { ThemeObject } from '@vuepress/core'
+import { path } from '@vuepress/utils'
+
+const fooTheme = {
```
Comment (meteorlxy) [inline docs/advanced/cookbook/extending-a-theme.md:33]: `const fooTheme: ThemeObject`

**Review by meteorlxy — CHANGES_REQUESTED** [review]

File: docs/advanced/cookbook/extending-a-theme.md (hunk 3 above)
Comment (YunYouJun) [reply docs/advanced/cookbook/extending-a-theme.md]: Done.

**Review by meteorlxy — APPROVED** [review]

**Pull request merged**